cli/runner.go

```diff
@@ -0,0 +1,305 @@
+	if err != nil {
+		return err
+	}
+
+	var cmd *exec.Cmd
+	if start {
+		cmd = exec.Command(executable, "up", "-d")
+	} else {
+		cmd = exec.Command(executable, "down")
+	}
+
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	log.Println("Executing", executable, "in the background...")
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	// if we want "silent" output we exit early
+	if quiet {
+		return nil
+	}
+
+	// otherwise see logs
+	log.Println("Printing logs:")
+	cmd = exec.Command(executable, "logs", "-f")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Find a compose executable
+func findExec(compose bool) (error, string) {
+	osPath := os.Getenv("PATH")
+
+	// I refuse to spend more time to debug this OS's blasphemy
+	if runtime.GOOS == "windows" {
+		if compose {
+			return nil, "docker-compose.exe"
+		} else {
+			return nil, "docker.exe"
+		}
+	}
+
+	osPathList := strings.Split(osPath, string(os.PathListSeparator))
+
+	podmanPresent := false
+	dockerPresent := false
+
+	for _, path := range osPathList {
+		files, err := ioutil.ReadDir(path)
+		if err != nil {
+			return err, ""
+		}
+
+		for _, f := range files {
+			if compose {
+				switch f.Name() {
+				case "podman-compose":
+					podmanPresent = true
+				case "docker-compose":
+					dockerPresent = true
+				}
+			} else {
+				switch f.Name() {
+				case "podman":
+					podmanPresent = true
+				case "docker":
+					dockerPresent = true
+				}
+			}
+		}
+	}
+
+	//podman takes precedence
+	if compose {
+		if podmanPresent {
+			return nil, "podman-compose"
+		}
+		if dockerPresent {
+			return nil, "docker-compose"
+		}
+	} else {
+		if podmanPresent {
+			return nil, "podman"
+		}
+		if dockerPresent {
+			return nil, "docker"
+		}
+	}
+
+	return errors.New("executable not found"), ""
+}
+
+type PodmanNetwork struct {
+	Name   string `json:"name"`
+	Id     string `json:"id"`
+	Driver string `json:"driver"`
+	// NetIF    string   `json:"network_interface"`
+	// Created  string   `json:"created"`
+	Subnets []Subnet `json:"subnets"`
+	// IPv6     bool     `json:"ipv6_enabled"`
+	// Internal bool     `json:"internal"`
+	// DNS      bool     `json:"dns_enabled"`
+	// IpamOpt  IpamOpt  `json:"ipam_options"`
+}
+
+type DockerNetwork struct {
+	Name   string `json:"Name"`
+	Id     string `json:"Id"`
+	Driver string `json:"Driver"`
+	// 	Scope    string   `json:"Scope"`
+	// 	Created  string   `json:"Created"`
+	// 	IPv6     bool     `json:"EnableIPv6"`
+	// 	Internal bool     `json:"Internal"`
+	// 	Attachable
+	// 	Ingress
+	// 	Containers
+	// 	DNS      bool     `json:"dns_enabled"`
+	Ipam Ipam `json:"IPAM"`
+}
+
+type Ipam struct {
+	//Driver
+	//Options
+	Config []Subnet
+}
+
+type Subnet struct {
+	Subnet  string `json:"subnet"`
+	Gateway string `json:"gateway"`
+}
+type IpamOpt struct {
+	Driver string `json:"driver"`
+}
+
+func GetCNIGateway() (error, string) {
```

Comment:
`error should be returned as the last argument ST1008`

Reply:
fixed